### ISSUE TITLE
Markdown reader: taking an unusually long time during a certain kind of input

### ISSUE THREAD

**Issue by DaedlyKitten** [jgm/pandoc#5166]
The following is a 104 KB txt file of merely repeating the same paragraph that after some trial and error I managed to isolate from a lengthy novel I initially wanted to convert into EPUB, 
when converting this sample txt to a EPUB file, it will take an exceedingly very long time, approximately  several minutes on my PC, when usually any other ~100kb txt file conversion takes less than 1 second.

This is the file and command I used:

`pandoc.exe "%homepath%\Downloads\sample.txt" -o %homepath%\downloads\11111.epub`

[sample.txt](https://github.com/jgm/pandoc/files/2693913/sample.txt)

version info:
pandoc.exe 2.5
Compiled with pandoc-types 1.17.5.4, texmath 0.11.1.2, skylighting 0.7.4
Default user data directory: C:\Users\konpo\AppData\Roaming\pandoc
Copyright (C) 2006-2018 John MacFarlane

**Comment by DaedlyKitten** [comment]
some further tries seem to suggest it has something to do with its particular length rather than specific characters? as the conversion time gets back to normal when i inserted random characters into the paragraph...

**Comment by mb21** [comment]
I can confirm even this with pandoc 2.5 takes way too long:

    pandoc -f markdown -t html https://github.com/jgm/pandoc/files/2693913/sample.txt

Yet it works fine with pandoc 2.4

**Comment by jgm** [comment]
If this is something that occurs in 2.5 but not
previous versions, then I suspect the following change
in the 2.5 chalnegolg:

> Fix parsing of citations, quotes, and underline
> emphasis after symbols. Starting with pandoc 2.4,
> citations, quoted inlines, and underline emphasis were
> no longer recognized after certain symbols, like
> parentheses (#5099, #5053).

This could be verified by using nightlies produced
from before and after these changes.

**Comment by mb21** [comment]
@jgm, you are right. Reverting edc651059ee617cf36b511de080a646d2e6513a4 fixes the performance degradation:

    git revert edc651059ee617cf36b511de080a646d2e6513a4

**Comment by jgm** [comment]
@DaedlyKitten if you use `-f markdown-smart`, you will not experience any performance degradation.
It happens because pandoc is trying to match up quotes and things like
```
'去岁嵩山之会
```
are causing the problem. It would be good to fix things so you don't get the slowdown even with `+smart`, but this workaround should be fine for your purposes.

**Comment by mb21** [comment]
> It would be good to fix things so you don't get the slowdown even with `+smart`

therefore reopening.

**Comment by jgm** [comment]
A similar example from #7306, using `pandoc -f textile`:
```
&#60;config&#62;
    &#60;section1&#62;
        &#60;a&#62;
                &#60;format identifier="1" offset="0" length="22" pn_format="%u-%u-%u" sn_format="%u-%u-%u-%u"&#62;%*02u%04u%04u%04u%02u%02u%03u%03u&#60;/format&#62;
        &#60;/a&#62;
    &#60;/section1&#62;
    &#60;section2&#62;
        &#60;b&#62;
                &#60;format identifier="2" offset="0" length="17" pn_format="%u-%u-%u" sn_format="%u-%u-%u-%u"&#62;%02u%03u%03u%02u%02u%02u%03u&#60;/format&#62;
        &#60;/b&#62;
    &#60;/section2&#62;
    &#60;section3&#62;
        &#60;c&#62;
                &#60;format identifier="2" offset="0" length="17" pn_format="%u-%u-%u" sn_format="%u-%u-%u-%u"&#62;%02u%03u%03u%02u%02u%02u%03u&#60;/format&#62;
        &#60;/c&#62;
    &#60;/section3&#62;
&#60;/config&#62;
```
